handler/oauth2/flow_generic_code_token.go

```diff
@@ -60,7 +75,12 @@ func (c *GenericCodeTokenEndpointHandler) PopulateTokenEndpointResponse(ctx cont
 		return errorsx.WithStack(fosite.ErrUnknownRequest)
 	}
 
-	code, signature, ar, err := c.GetCodeAndSession(ctx, requester)
+	code, signature, err := c.Code(ctx, requester)
```

Comment:
`err` gets redefined over and over, i m surprised if this compiles

try to define the variables u need

 ```
var code <type>
var signature <type>
var ar <type>
 ```
 
 and then use the `err` in closure within the if clauses, otherwise define `err` as a var too and drop the `:=` for the `=`